cpp/src/reductions/simple.cuh

```diff
@@ -46,29 +53,43 @@ namespace simple {
  * @return Output scalar in device memory
  */
 template <typename ElementType, typename ResultType, typename Op>
-std::unique_ptr<scalar> simple_reduction(column_view const& col,
-                                         rmm::cuda_stream_view stream,
-                                         rmm::mr::device_memory_resource* mr)
+scalar_or_column_ptr simple_reduction(
+  column_view const& col,
+  std::optional<std::reference_wrapper<column_view const>> offsets,
```

Comment:
I'm having trouble supporting the idea of a segmented reduction and a normal reduction sharing the same code path. They are different algorithms with different implementations. This feels like a single responsibility violation at best and plain confusing at worst.

Reply:
Yeah now that I'm seeing the practical application that you had in mind during our discussion I am also not convinced. Reducing LOC is nice, but these are fundamentally different and all you're gaining is to avoid duplicating a few levels of dispatch.